Concatenate Issue when receiving a condition array

```python
import numpy as np


#pythran export outputCodeActiveZoneOpt()
def outputCodeActiveZoneOpt():
    din = [1, 2, 56, 13, 3, 45, 76, 7345, 34, 67, 5]
    a = np.array(din, dtype=np.int64)
    C1 = a < 50
    b = a[C1]
    C2 = np.concatenate([[True], (np.diff(b) != 1)])
    return C2
```

gives 

`In file included from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/__builtin__/bool_.hpp:5:0,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/__builtin__/bool_.hpp:4,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/NoneType.hpp:8,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/slice.hpp:5,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/core.hpp:29,
                 from /tmp/tmpj87jd7kg.cpp:1:
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp: In instantiation of ‘{anonymous}::pythonic::types::array<T, sizeof... (S)> {anonymous}::pythonic::types::_to_array(const Tuple&, {anonymous}::pythonic::utils::index_sequence<N ...>) [with T = {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >; Tuple = std::tuple<{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >; long unsigned int ...S = {0, 1}; {anonymous}::pythonic::types::array<T, sizeof... (S)> = {anonymous}::pythonic::types::array_base<{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >, 2, {anonymous}::pythonic::types::tuple_version>; {anonymous}::pythonic::utils::index_sequence<N ...> = {anonymous}::pythonic::utils::integer_sequence<long unsigned int, 0, 1>]’:
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp:577:24:   required from ‘{anonymous}::pythonic::types::array<T, sizeof... (Tys)> {anonymous}::pythonic::types::to_array(const std::tuple<_Elements ...>&) [with T = {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >; Tys = {{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> >}; {anonymous}::pythonic::types::array<T, sizeof... (Tys)> = {anonymous}::pythonic::types::array_base<{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >, 2, {anonymous}::pythonic::types::tuple_version>]’
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/__builtin__/pythran/static_list.hpp:36:65:   required from ‘{anonymous}::pythonic::types::static_list<typename __combined<T0, Tys ...>::type, (1 + sizeof... (Tys))> {anonymous}::pythonic::__builtin__::pythran::static_list(const std::tuple<_El0, _El ...>&) [with T0 = {anonymous}::pythonic::types::list<bool>; Tys = {{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> >}; {anonymous}::pythonic::types::static_list<typename __combined<T0, Tys ...>::type, (1 + sizeof... (Tys))> = {anonymous}::pythonic::types::array_base<{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >, 2, {anonymous}::pythonic::types::list_version>; typename __combined<T0, Tys ...>::type = {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >]’
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/__builtin__/pythran/static_list.hpp:39:5:   required from ‘decltype ({anonymous}::pythonic::__builtin__::pythran::static_list((forward<Types>)(<unnamed>::pythonic::__builtin__::pythran::functor::static_list::operator()::types)...)) {anonymous}::pythonic::__builtin__::pythran::functor::static_list::operator()(Types&& ...) const [with Types = {std::tuple<{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >}; decltype ({anonymous}::pythonic::__builtin__::pythran::static_list((forward<Types>)(<unnamed>::pythonic::__builtin__::pythran::functor::static_list::operator()::types)...)) = {anonymous}::pythonic::types::array_base<{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >, 2, {anonymous}::pythonic::types::list_version>]’
/tmp/tmpj87jd7kg.cpp:64:512:   required from here
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp:571:14: error: no matching function for call to ‘{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >::numpy_expr(std::__tuple_element_t<0, std::tuple<{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > > >&)’
     return {{static_cast<T>(std::get<S>(t))...}};
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/ndarray.hpp:39:0,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/tuple.hpp:12,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/__builtin__/bool_.hpp:7,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/NoneType.hpp:8,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/slice.hpp:5,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/core.hpp:29,
                 from /tmp/tmpj87jd7kg.cpp:1:
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/numpy_expr.hpp:73:3: note: candidate: {anonymous}::pythonic::types::numpy_expr<O, Args>::numpy_expr(const Args& ...) [with Op = {anonymous}::pythonic::operator_::functor::add; Args = {{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> >}]
   numpy_expr<Op, Args...>::numpy_expr(Args const &... args)
   ^~~~~~~~~~~~~~~~~~~~~~~
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/numpy_expr.hpp:73:3: note:   candidate expects 2 arguments, 1 provided
In file included from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:36:0,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/ndarray.hpp:4,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/tuple.hpp:12,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/__builtin__/bool_.hpp:7,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/NoneType.hpp:8,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/slice.hpp:5,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/core.hpp:29,
                 from /tmp/tmpj87jd7kg.cpp:1:
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/numpy_expr.hpp:575:5: note: candidate: constexpr {anonymous}::pythonic::types::numpy_expr<O, Args>::numpy_expr({anonymous}::pythonic::types::numpy_expr<O, Args>&&) [with Op = {anonymous}::pythonic::operator_::functor::add; Args = {{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> >}]
     numpy_expr(numpy_expr &&) = default;
     ^~~~~~~~~~
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/numpy_expr.hpp:575:5: note:   no known conversion for argument 1 from ‘std::__tuple_element_t<0, std::tuple<{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > > > {aka const {anonymous}::pythonic::types::list<bool>}’ to ‘{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >&&’
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/numpy_expr.hpp:574:5: note: candidate: {anonymous}::pythonic::types::numpy_expr<O, Args>::numpy_expr(const {anonymous}::pythonic::types::numpy_expr<O, Args>&) [with Op = {anonymous}::pythonic::operator_::functor::add; Args = {{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> >}]
     numpy_expr(numpy_expr const &) = default;
     ^~~~~~~~~~
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/numpy_expr.hpp:574:5: note:   no known conversion for argument 1 from ‘std::__tuple_element_t<0, std::tuple<{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > > > {aka const {anonymous}::pythonic::types::list<bool>}’ to ‘const {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >&’
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/numpy_expr.hpp:573:5: note: candidate: constexpr {anonymous}::pythonic::types::numpy_expr<O, Args>::numpy_expr() [with Op = {anonymous}::pythonic::operator_::functor::add; Args = {{anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> >}]
     numpy_expr() = default;
     ^~~~~~~~~~
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/numpy_expr.hpp:573:5: note:   candidate expects 0 arguments, 1 provided
In file included from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/__builtin__/bool_.hpp:5:0,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/__builtin__/bool_.hpp:4,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/NoneType.hpp:8,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/types/slice.hpp:5,
                 from /home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/core.hpp:29,
                 from /tmp/tmpj87jd7kg.cpp:1:
/home/igajadua/.local/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp:571:48: error: could not convert ‘{{<expression error>}}’ from ‘<brace-enclosed initializer list>’ to ‘{anonymous}::pythonic::types::array<{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >, 2> {aka {anonymous}::pythonic::types::array_base<{anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::add, {anonymous}::pythonic::types::list<bool>, {anonymous}::pythonic::types::numpy_expr<{anonymous}::pythonic::operator_::functor::ne, {anonymous}::pythonic::types::ndarray<long int, {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version> >, {anonymous}::pythonic::types::broadcast<long int, long int> > >, 2, {anonymous}::pythonic::types::tuple_version>}’
     return {{static_cast<T>(std::get<S>(t))...}};
                                                ^
WARNING: Compilation error, trying hard to find its origin...
WARNING: Nop, I'm going to flood you with C++ errors!
CRITICAL: Cover me Jack. Jack? Jaaaaack!!!!
`

I have simplified it all I could.

Regards,
Rodrigo Iga.

--- a/pythran/optimizations/list_to_tuple.py
+++ b/pythran/optimizations/list_to_tuple.py
@@ -10,6 +10,7 @@ import gast as ast
 patterns = (MODULES['numpy']['ones'],
             MODULES['numpy']['zeros'],
             MODULES['numpy']['empty'],
+            MODULES['numpy']['concatenate'],
             MODULES['__builtin__']['tuple'],
             )
 

--- a/pythran/tests/test_numpy_func2.py
+++ b/pythran/tests/test_numpy_func2.py
@@ -312,6 +312,9 @@ def test_copy0(x):
             numpy.array([[1, 2], [3, 4]]),
             np_concatenate2=[NDArray[int,:,:]])
 
+    def test_concatenate3(self):
+        self.run_test("def np_concatenate3(a): from numpy import array, concatenate ; return concatenate([[1],a + a])", numpy.array([1, 2]), np_concatenate3=[NDArray[int,:]])
+
     def test_hstack0(self):
         self.run_test("def np_hstack0(a,b): import numpy as np; return np.hstack((a,b))",
                       numpy.array((1,2,3)),

